src/Network/HTTPRequest.php

```diff
@@ -103,120 +105,136 @@ public function get(string $url, array $opts = [], &$redirects = 0)
 			return CurlResult::createErrorCurl($url);
 		}
 
-		$ch = @curl_init($url);
-
-		if (($redirects > 8) || (!$ch)) {
-			$this->profiler->stopRecording();
-			return CurlResult::createErrorCurl($url);
-		}
+		$curlOptions = [];
 
-		@curl_setopt($ch, CURLOPT_HEADER, true);
+		$curlOptions[CURLOPT_HEADER] = true;
 
 		if (!empty($opts['cookiejar'])) {
-			curl_setopt($ch, CURLOPT_COOKIEJAR, $opts["cookiejar"]);
-			curl_setopt($ch, CURLOPT_COOKIEFILE, $opts["cookiejar"]);
+			$curlOptions[CURLOPT_COOKIEJAR] = $opts["cookiejar"];
+			$curlOptions[CURLOPT_COOKIEFILE] = $opts["cookiejar"];
 		}
 
 		// These settings aren't needed. We're following the location already.
-		//	@curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
-		//	@curl_setopt($ch, CURLOPT_MAXREDIRS, 5);
+		//	$curlOptions[CURLOPT_FOLLOWLOCATION] =true;
+		//	$curlOptions[CURLOPT_MAXREDIRS] = 5;
 
 		if (!empty($opts['accept_content'])) {
-			curl_setopt(
-				$ch,
-				CURLOPT_HTTPHEADER,
-				['Accept: ' . $opts['accept_content']]
-			);
+			if (empty($curlOptions[CURLOPT_HTTPHEADER])) {
+				$curlOptions[CURLOPT_HTTPHEADER] = [];
+			}
```

Comment:
Please initialize `$curlOptions[CURLOPT_HTTPHEADER]` to an empty list outside of this condition, which saves you having to check its existence thrice in this method.

Reply:
done, I wasn't sure if setting this option to `null` differs from empty `[]`